src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -13,7 +13,23 @@ namespace NuGet.PackageManagement.VisualStudio
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class VSSettings : Configuration.ISettings
     {
-        private Configuration.ISettings SolutionSettings { get; set; }
+        // to initialize SolutionSettings first time outside MEF constructor
+        private Configuration.ISettings _solutionSettings;
+
+        private Configuration.ISettings SolutionSettings
```

Comment:
`Another problem here is ResetSolutionSettings has to run on UI thread due to number of DTE calls.`

Where are these DTE calls? I don't see anything in that API.